src/tests/test_generic_server.py

```diff
@@ -0,0 +1,165 @@
+            with mock.patch('generic_server.GenericServer._reinit') as _reinit:
+                self.svr._reinit2()
+                log_str = str(log.mock_calls)
+                self.assertTrue(log_contents in log_str, log_str)
+                self.assertTrue(_reinit.called)
+
+    def test__reinit(self):
+        with mock.patch('hostaddr.update_domains'):
+            with mock.patch('generic_server.GenericServer.reinit') as reinit:
+                with mock.patch('Trace.log') as log:
+                    self.svr._reinit()
+                    self.assertTrue(reinit.called)
+                    self.assertTrue(log.called)
+
+    def test___init__(self):
+        self.assertTrue(isinstance(self.svr, generic_client.GenericClient))
+        self.assertTrue(isinstance(self.svr, generic_server.GenericServer))
+
+    def test_handle_generic_commands(self):
+        gsi = generic_server.GenericServerInterface()
+        with mock.patch('Trace.do_print') as do_print:
+            with mock.patch('Trace.dont_print')as dont_print:
+                with mock.patch('Trace.do_log') as do_log:
+                    with mock.patch('Trace.dont_log') as dont_log:
+                        with mock.patch('Trace.do_alarm') as do_alarm:
+                            with mock.patch('Trace.dont_alarm') as dont_alarm:
+                                self.svr.handle_generic_commands(gsi)
+                                self.assertFalse(do_print.called)
+                                self.assertFalse(dont_print.called)
+                                self.assertFalse(do_log.called)
+                                self.assertFalse(dont_log.called)
+                                self.assertFalse(do_alarm.called)
+                                self.assertFalse(dont_alarm.called)
+                                gsi.do_print.append('test')
+                                gsi.dont_print.append('test')
+                                gsi.do_log.append('test')
+                                gsi.dont_log.append('test')
+                                gsi.do_alarm.append('test')
+                                gsi.dont_alarm.append('test')
+                                self.svr.handle_generic_commands(gsi)
+                                self.assertTrue(do_print.called)
+                                self.assertTrue(dont_print.called)
+                                self.assertTrue(do_log.called)
+                                self.assertTrue(dont_log.called)
+                                self.assertTrue(do_alarm.called)
+                                self.assertTrue(dont_alarm.called)
+
+    def test_get_log_name(self):
+        logn = self.svr.get_log_name('foo.bar')
+        self.assertEqual(logn, 'FOO.BAR')
+        logn = self.svr.get_log_name('foo.bar.baz')
+        self.assertEqual(logn, 'FOO.BAR.BAZ')
+
+    def test_get_name(self):
+        nm = self.svr.get_name('foo')
+        self.assertEqual(nm, 'FOO', nm)
```

Comment:
Is the third arg here intentional?

Reply:
It was intentional.  Changed it to self.assertEqual(nm, 'FOO', "expected %s=='FOO'" % nm) for better readability